### PR TITLE
Added UUID case to GetTypeToPython

### LINKED ISSUE #5884: Python connection description fails for UUID columns
### What happens?

When getting a description for a connection that has query results with a UUID column, it fails in python.

### To Reproduce

```py
import duckdb
con = duckdb.connect(database=':memory:')
con.execute("select uuid();")
con.description
```
Results in this error:
```
Traceback (most recent call last):
  ...
duckdb.NotImplementedException: Not implemented Error: Unsupported type: "UUID"
```

### OS:

macOS, Linux, x86

### DuckDB Version:

0.5.1

### DuckDB Client:

Python

### Full Name:

Mac Lockard

### Affiliation:

Hex

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by maclockard**
Fixes https://github.com/duckdb/duckdb/issues/5884

I have not done any testing myself that this fixes the issue, would not be surprised if there are other places that the UUID type needs to be handled

### CODE REVIEW

File: tools/pythonpkg/src/pyresult.cpp [hunk 1]
```diff
@@ -326,6 +326,7 @@ py::str GetTypeToPython(const LogicalType &type) {
 	}
 	case LogicalTypeId::JSON:
 	case LogicalTypeId::VARCHAR:
+	case LogicalTypeId::UUID:
 		return py::str("STRING");
```
Comment (Tishj) [inline tools/pythonpkg/src/pyresult.cpp:330]: Ideally this should be the Python type that this duckdb type converts to, which for UUID is not a string, but a `uuid.UUID` so can we change this to "UUID" for this case instead?

Comment (maclockard) [reply tools/pythonpkg/src/pyresult.cpp:330]: sure thing!

**Comment by Tishj** [comment]
I had one comment, but otherwise it looks good to me
For good measure could you add a test for this to `tools/pythonpkg/tests/fast/test_result.py` ?

**Comment by maclockard** [comment]
@Tishj Added a test case for basically if it throws or not. Are there any additional asserts you think are needed?

File: tools/pythonpkg/tests/fast/test_result.py [hunk 2]
```diff
@@ -50,3 +50,8 @@ def test_result_interval(self):
         res = rel.execute()
         assert res.description == [('ivals', 'TIMEDELTA', None, None, None, None, None)]
         assert res.fetchall() == [(datetime.timedelta(days=1.0),), (datetime.timedelta(seconds=2.0),), (datetime.timedelta(microseconds=1.0),)]
+
+    def test_description_uuid(self):
+        connection = duckdb.connect()
+        connection.execute("select uuid();")
+        connection.description
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/test_result.py:57]: Can you please add an assertion on what is being returned here?

**Review by Mause — CHANGES_REQUESTED** [review]

**Comment by Mytherin** [comment]
@maclockard could you have a look at addressing the test issue? LGTM otherwise.

**Pull request merged**